### PR TITLE
Fix bug when an html element has id='jquery'

### LINKED ISSUE #6193: `cy.visit()` Error - DOM Element `id` Attribute Set to `jquery`
### Current behavior:

Given a static webpage with zero JavaScript present (no jQuery library), if there exists a DOM element with an `id` attribute set to `jquery` (for example, `<p id="jquery">jQuery</p>`), then the `cy.visit()` method throws a TypeError: `TypeError: item.first is not a function.`

Debug Logs:
```
cypress_runner.js:159529 TypeError: item.first is not a function

Because this error occurred during a 'before each' hook we are skipping the remaining tests in the current suite: 'homepage'
    at Object.unwrapFirst (http://localhost:5000/__cypress/runner/cypress_runner.js:86108:19)
    at http://localhost:5000/__cypress/runner/cypress_runner.js:80941:33
From previous event:
    at runCommand (http://localhost:5000/__cypress/runner/cypress_runner.js:80933:8)
    at next (http://localhost:5000/__cypress/runner/cypress_runner.js:81014:14)
From previous event:
    at http://localhost:5000/__cypress/runner/cypress_runner.js:81055:37
From previous event:
    at run (http://localhost:5000/__cypress/runner/cypress_runner.js:81048:19)
    at Object.cy.<computed> [as visit] (http://localhost:5000/__cypress/runner/cypress_runner.js:81457:11)
    at Context.runnable.fn (http://localhost:5000/__cypress/runner/cypress_runner.js:81699:24)
    at callFn (http://localhost:5000/__cypress/runner/cypress_runner.js:30929:21)
    at Hook.../driver/node_modules/mocha/lib/runnable.js.Runnable.run (http://localhost:5000/__cypress/runner/cypress_runner.js:30922:7)
    at http://localhost:5000/__cypress/runner/cypress_runner.js:84876:28
From previous event:
    at Object.onRunnableRun (http://localhost:5000/__cypress/runner/cypress_runner.js:84864:20)
    at $Cypress.action (http://localhost:5000/__cypress/runner/cypress_runner.js:78026:61)
    at Hook.Runnable.run (http://localhost:5000/__cypress/runner/cypress_runner.js:83558:13)
    at next (http://localhost:5000/__cypress/runner/cypress_runner.js:31272:10)
    at http://localhost:5000/__cypress/runner/cypress_runner.js:31294:5
    at timeslice (http://localhost:5000/__cypress/runner/cypress_runner.js:26362:27)
logError @ cypress_runner.js:159529
```

Added a breakpoint to view the object that's considered a "jQuery object:"

### Desired behavior:

In a previous issue, this type of error was thrown when an older version of jQuery was found on the webpage being tested: https://github.com/cypress-io/cypress/issues/2927. According to the issue thread, it was resolved in Cypress v3.2.0. 

I'd expect that a webpage with zero JavaScript present shouldn't encounter this error.

### Test code to reproduce

https://github.com/kenchandev/cypress-bug-reduced-test-case

The `README.md` file will contain directions on how to quickly reproduce this bug.

### Versions

**Cypress**: v3.8.2
**Operating System**: Mac OS X
**Browser**: Cypress Test Runner's Default Browser

### PR DESCRIPTION
**Pull request by panzarino**
- Closes #6193

### User facing changelog

- Fixes an error where a page would not render if an DOM element had `"jquery"` as its `id`

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub?

### CODE REVIEW

**Comment by panzarino** [comment]
@Bkucera Should I add a separate test? I added `<p id="jquery">jQuery</p>` to an existing test html file. Before updating the driver, all of the tests that used that file would fail because they would not render. I'm not sure what to add as a test aside from `cy.visit()` which is already accomplished within existing tests for that file. Alternatively, I could create a separate html file and for this test but that seems unnecessary.

**Comment by Bkucera** [comment]
@panzarino good point, I would add a comment in the HTML file above the element along the lines of:
```
cy.visit() fails when element with id="jquery" exists
// https://github.com/cypress-io/cypress/issues/6193
``

**Comment by panzarino** [comment]
@Bkucera are you looking for something like the following:

```javascript
if (hasJqueryProperty) {
    const prop = _.get(obj, 'constructor.prototype.jquery')
    hasJqueryProperty = !!prop && prop.nodeType !== 1 && typeof prop.nodeName !== 'string'
}

return hasJqueryProperty
```

**Comment by Bkucera** [comment]
@panzarino actually, i dont think we need a function, but can you add a comment over the line you added explaining why all those checks are being made (e.g. `// NOTE: ...`)

**Comment by jennifer-shehane** [comment]
@Bkucera Is this good to go? I manually confirmed this fixes the original issue.

File: packages/driver/src/dom/jquery.js [hunk 1]
```diff
@@ -31,7 +31,14 @@ const isJquery = (obj) => {
...
   // on its prototype?
-  return hasJqueryProperty && !!_.get(obj, 'constructor.prototype.jquery')
+  const prop = _.get(obj, 'constructor.prototype.jquery')
+
+  // an HTML element with id="jquery" will show up
+  // as the jquery property of the window constructor
+  // note: instanceof HTMLElement is ineffective
+  // on objects from a different window/frame
+  // so the nodeType and nodeName is used to check if it is an HTML element
+  return hasJqueryProperty && !!prop && prop.nodeType !== 1 && typeof prop.nodeName !== 'string'
```
Comment (Bkucera) [inline packages/driver/src/dom/jquery.js:41]: can you explain why `obj && ('jquery' in obj || obj.jquery)` is evaluating to true? That's what was causing the bug right?

Comment (panzarino) [reply packages/driver/src/dom/jquery.js:41]: For any element, the id becomes a property of the window: https://2ality.com/2012/08/ids-are-global.html.

In this case, we will falsely detect that `window.jquery` is the jquery library when it is actually this element.

Comment (Bkucera) [reply packages/driver/src/dom/jquery.js:41]: can we just test if `hasJqueryProperty && typeof prop === 'string'` ?

Comment (panzarino) [reply packages/driver/src/dom/jquery.js:41]: Is prop a string when it is actually jquery? I can run some tests tomorrow

Comment (Bkucera) [reply packages/driver/src/dom/jquery.js:41]: I believe its the version number string

Comment (panzarino) [reply packages/driver/src/dom/jquery.js:41]: Oh yeah I do remember that, I guess that would work as well if you would rather it be that way

Comment (Bkucera) [reply packages/driver/src/dom/jquery.js:41]: makes more sense to me, if we know .jquery must be a string no sense testing if it's not an element

**Comment by rramji0211** [comment]
Hello there, wanted to know when this will be released please.

**Comment by jennifer-shehane** [comment]
@rramji0211 We're working on the release now. There are [multiple ways to see when there's a release](https://on.cypress.io/general-questions-faq#How-can-I-find-out-about-new-versions-of-Cypress), but we'll specifically comment on the closed issue when it's released also.

**Pull request merged**